torchx/tracker/api.py

```diff
@@ -181,26 +181,27 @@ def build_trackers(
 ) -> Iterable[TrackerBase]:
     trackers = []
 
-    entrypoint_factories = load_group("torchx.tracker")
+    entrypoint_factories = load_group("torchx.tracker") or {}
     if not entrypoint_factories:
-        logger.warning(
-            "No 'torchx.tracker' entry_points are defined. Tracking will not capture any data."
-        )
-        return trackers
+        logger.warning("No 'torchx.tracker' entry_points are defined.")
 
     for entrypoint_key, config in entrypoint_and_config.items():
-        if entrypoint_key not in entrypoint_factories:
-            logger.warning(
-                f"Could not find `{entrypoint_key}` tracker entrypoint. Skipping..."
+        factory = entrypoint_factories.get(entrypoint_key)
+        if factory is None:
+            logger.info(
+                f"Could not find `{entrypoint_key}` among entry_points, checking modules..."
             )
-            continue
-        factory = entrypoint_factories[entrypoint_key]
+            factory = load_module(entrypoint_key)
+            if not factory:
+                logger.warning(
+                    f"Could not find `{entrypoint_key}` tracker entrypoint. Skipping..."
+                )
+                continue
```

Comment:
The logging here should be cleaned up. Rather than logging that you "couldn't" find something, we should run through the fallback logic (first check `entry_points` then `modules`) and at the end, log the result. Something to the effect of:

```
_src = {entry_point or module}

if factory:
   logger.info("Tracker factory `{factory_name}` found in `{_src}`. Creating tracker with config={config}`)
else:
   logger.warning("No tracker factory `{factory_name}` found in entry_point or modules. See: https://pytorch.org/torchx/main/tracker.html#module-torchx.tracker")

```